### PR TITLE
Fix constraint validation in components

### LINKED ISSUE #5678: fix: constraint validation API in form controls doesn't work and prevents form submission
# 🐛 Bug Report

When using `required`, `pattern`, and other attributes associated with the [constraint validation API](https://developer.mozilla.org/en-US/docs/Web/API/Constraint_validation), an error is thrown and no constraint is signaled when the form is submitted.

## 💻 Repro or Code Sample

https://jsfiddle.net/v7cmd0Lj/

## 🤔 Expected Behavior

The browser should show an alert stating that a value is required in the form control and prevent submission.

## 😯 Current Behavior

In Edge, Chrome, and Safari, the following error is received when submitting the form:

> An invalid form control with name='name' is not focusable.

In Firefox, no error is thrown, no validation message is shown, and the form does not submit.

## 💁 Possible Solution

[This answer on StackOverflow](https://stackoverflow.com/a/28340579/567486) describes what's happening in more detail.

## 🔦 Context

Our users have filed bug reports expecting `required` and similar constraint validation attributes to work as documented.

## 🌍 Your Environment

* Browser Edge 98, Chrome 98, Safari 15.3
* Version [e.g. 1.8.0]

### PR DESCRIPTION
**Pull request by nicholasrice**
# Pull Request

## 📖 Description
This PR adds an optional anchor argument to `FormAssociated.validate()`, which gets provided to `ElementInternals.setValidity()`. This anchor is the element to which the browser should focus within a shadowRoot if the element delegates focus. 
All components that implement `FormAssociated` *and* `delegatesFocus` have been updated to invoke validate() with the shadowed element that the element delegates focus to.

### 🎫 Issues
fixes #5678

## 👩‍💻 Reviewer Notes
The linked issue offers a brief discussion around potentially using `@open-wc/form-control` package to address this issue. I've opened a separate discussion for that conversation in #6300. For now, I'd like to fix the bug and we can take migration as a separate consideration and task.

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Review by scomea — APPROVED** [review]

File: packages/web-components/fast-foundation/src/number-field/number-field.ts [hunk 1]
```diff
@@ -235,6 +235,10 @@ export class FASTNumberField extends FormAssociatedNumberField {
         this.isUserInput = false;
     }
 
+    public validate(): void {
```
Comment (scomea) [inline packages/web-components/fast-foundation/src/number-field/number-field.ts:238]: Since these functions are public should they have descriptions?

Comment (nicholasrice) [reply packages/web-components/fast-foundation/src/number-field/number-field.ts:239]: Yep good call, updated to inherit docs from FormAssociated

**Review by EisenbergEffect — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]
Thanks @nicholasrice - do you think this is one we can cherry-pick into the `archives/fast-element-1`?

**Comment by nicholasrice** [comment]
@chrisdholt Yeah that is a good idea. I'll kick off a separate PR.

**Pull request merged**